lisa/tools/virtualclient.py

```diff
@@ -0,0 +1,398 @@
+
+        return "".join(cmd_parts)
+
+    def download_raw_data_file(self) -> Path:
+        local_path = self.node.local_log_path / "metrics.csv"
+        remote_path = self.node.get_pure_path(f"{self._vc_log_path}/metrics.csv")
+        current_user = self.node.tools[Whoami].get_username()
+        self.node.tools[Chown].change_owner(remote_path, current_user)
+        self.node.shell.copy_back(
+            remote_path,
+            local_path,
+        )
+        return local_path
+
+    def send_metrics_message(self, metrics_file: Path) -> None:
+        with open(metrics_file, mode="r", newline="") as csvfile:
+            reader = csv.DictReader(csvfile)
+            metric_messages: List[VCMetrics] = []
+            for row in reader:
+                metric = VCMetrics(
+                    time_stamp=self._parse_timestamp(row["Timestamp"]),
+                    experiment_id=row["ExperimentId"],
+                    client_id=row["ClientId"],
+                    profile=row["Profile"],
+                    profile_name=row["ProfileName"],
+                    tool_name=row["ToolName"],
+                    scenario_name=row["ScenarioName"],
+                    scenario_start_time=self._parse_timestamp(row["ScenarioStartTime"]),
+                    scenario_end_time=self._parse_timestamp(row["ScenarioEndTime"]),
+                    metric_categorization=row["MetricCategorization"],
+                    metric_name=row["MetricName"],
+                    metric_value=Decimal(row["MetricValue"]),
+                    metric_unit=row["MetricUnit"],
+                    metric_description=row["MetricDescription"],
+                    metric_relativity=row["MetricRelativity"],
+                    execution_system=row["ExecutionSystem"],
+                    operating_system_platform=row["OperatingSystemPlatform"],
+                    operation_id=row["OperationId"],
+                    operation_parent_id=row["OperationParentId"],
+                    app_host=row["AppHost"],
+                    app_name=row["AppName"],
+                    app_version=row["AppVersion"],
+                    app_telemetry_version=row["AppTelemetryVersion"],
+                    tags=row["Tags"],
+                )
+                metric_messages.append(metric)
+        for metri_messages in metric_messages:
+            notifier.notify(metri_messages)
+
+    def _parse_timestamp(self, timestamp_str: str) -> datetime:
+        # Strip 'Z' and limit microseconds to 6 digits
+        timestamp_str = timestamp_str.rstrip("Z")
+        # Split timestamp and microseconds
+        if "." in timestamp_str:
+            main_time, microseconds = timestamp_str.split(".")
+            # Truncate microseconds to 6 digits
+            microseconds = microseconds[:6]
+            timestamp_str = f"{main_time}.{microseconds}"
+        return datetime.strptime(timestamp_str, "%Y-%m-%dT%H:%M:%S.%f")
+
+
+class WorkloadRunner(Tool):
```

Comment:
This class doesn't need to be inherited from tools. It can accept multiple nodes as a construct parameter. So the vc client initialization is done by this class.